packages/microcosm/src/history.js

```diff
@@ -336,13 +332,15 @@ class History extends Emitter {
     while (size > this.limit && root.complete) {
       size -= 1
 
-      this._emit('remove', root.parent)
-
       if (root.next) {
         root = root.next
       }
     }
 
+    if (root.parent) {
+      this._emit('remove', root.parent.parent, null)
```

Comment:
the combination of `this._emit('remove', action, action)` and `this._emit('remove', root.parent.parent, null)` seems like a pretty funky code smell

Reply:
Hah. Yeah.. you're right...

Reply:
likely tied to the smelly implementation of removeSnapshotRange.